pkg/apply/task/inv_add_task.go

```diff
@@ -38,30 +49,82 @@ func (i *InvAddTask) Action() event.ResourceAction {
 }
 
 func (i *InvAddTask) Identifiers() object.ObjMetadataSet {
-	return object.UnstructuredSetToObjMetadataSet(i.Objects)
+	return object.UnstructuredSetToObjMetadataSet(i.ApplyObjects).Union(
```

Comment:
This refactor to pass in ApplyObjects and PruneObjects, instead of just Objects might be extractable. But I'm not sure if it makes sense by itself or not.